File: src/get_seclargest.py
def second_largest(arr):
    """
    Python implementation mimicking the exact logic and edge-case
    handling (-1 returns) of the original C++ function.
    """
    # Using 32-bit INT_MIN to match C++ behavior
    INT_MIN = -2147483648
    largest = INT_MIN
    second = INT_MIN

    for num in arr:
        if num > largest:
            second = largest
            largest = num
        elif num > second and num != largest:
            second = num

    return -1 if second == INT_MIN else second

File: src/get_seclargest.cpp
#include <iostream>
#include <vector>
#include <climits>

int secondLargest(const std::vector<int>& arr) {
    int largest = INT_MIN;
    int second = INT_MIN;

    for (int num : arr) {
        if (num > largest) {
            second = largest;
            largest = num;
        } else if (num > second && num != largest) {
            second = num;
        }
    }

    return second == INT_MIN ? -1 : second;
}

int main() {
    int n;
    std::cin >> n;

    std::vector<int> arr(n);
    for (int i = 0; i < n; i++) {
        std::cin >> arr[i];
    }

    std::cout << secondLargest(arr);
    return 0;
}